### PR TITLE
iterate over all links and calculate the CoG of the MAV

### PR DESCRIPTION
**Pull request by burrimi**

### CODE REVIEW

File: rotors_gazebo_plugins/src/gazebo_multirotor_base_plugin.cpp [hunk 1]
```diff
@@ -59,9 +59,29 @@ void GazeboMultirotorBasePlugin::Load(physics::ModelPtr _model, sdf::ElementPtr
...
   child_links_ = link_->GetChildJointsLinks();
+
+  double total_mass = 0;
+  math::Vector3 weighted_position = math::Vector3::Zero;
+  double mass = link_->GetInertial()->GetMass();
+  total_mass = mass;
+  weighted_position = link_->GetRelativePose().pos * mass;
+
+  std::cout << "[Multirotor base]: Calculating CoG" << std::endl;
+  std::cout << link_->GetScopedName() << " CoG: " << link_->GetRelativePose() << " Mass: ";
```
Comment (ffurrer) [inline rotors_gazebo_plugins/src/gazebo_multirotor_base_plugin.cpp]: I think you should also add:
```C++
link_->GetCog()
```
to the relative pose?

File: rotors_gazebo_plugins/src/gazebo_multirotor_base_plugin.cpp [hunk 2]
```diff
@@ -59,9 +59,29 @@ void GazeboMultirotorBasePlugin::Load(physics::ModelPtr _model, sdf::ElementPtr
...
+  std::cout << link_->GetInertial()->GetMass() << std::endl;
+
   for (unsigned int i = 0; i < child_links_.size(); i++) {
     std::string link_name = child_links_[i]->GetScopedName();
 
+    math::Vector3 position = child_links_[i]->GetRelativePose().pos;
```
Comment (ffurrer) [inline rotors_gazebo_plugins/src/gazebo_multirotor_base_plugin.cpp]: same here

File: rotors_gazebo_plugins/src/gazebo_multirotor_base_plugin.cpp (hunk 1 above)
Comment (ffurrer) [inline rotors_gazebo_plugins/src/gazebo_multirotor_base_plugin.cpp]: ahh maybe it's `link_->GetInertial()->GetCog()` or something alike :)

File: rotors_gazebo_plugins/src/gazebo_multirotor_base_plugin.cpp [hunk 3]
```diff
@@ -59,9 +59,29 @@ void GazeboMultirotorBasePlugin::Load(physics::ModelPtr _model, sdf::ElementPtr
       boost::bind(&GazeboMultirotorBasePlugin::OnUpdate, this, _1));
 
   child_links_ = link_->GetChildJointsLinks();
+
+  double total_mass = 0;
+  math::Vector3 weighted_position = math::Vector3::Zero;
+  double mass = link_->GetInertial()->GetMass();
+  total_mass = mass;
+  weighted_position = link_->GetRelativePose().pos * mass;
```
Comment (ffurrer) [inline rotors_gazebo_plugins/src/gazebo_multirotor_base_plugin.cpp]: ahh and here the cog should also be added

**Comment by ffurrer** [comment]
scolgtm

**Comment by ffurrer** [comment]
@burrimi ping

**Comment by burrimi** [comment]
Will include that in new PR.

**Pull request closed without merging**